src/complex/DataStructure/DataStructure.cpp

```diff
@@ -167,17 +169,24 @@ Result<LinkedPath> DataStructure::makePath(const DataPath& path)
     if(data == nullptr)
     {
       data = complex::DataGroup::Create(*this, name);
+      createdIds.push_back(data->getId());
     }
     const BaseGroup* parent = dynamic_cast<const BaseGroup*>(data);
     pathIds.push_back(data->getId());
 
     for(usize i = 1; i < path.getLength(); i++)
     {
+      if(parent == nullptr)
+      {
+        return complex::MakeErrorResult<LinkedPath>(-3, "Target parent object in path is not derived from BaseGroup.");
```

Comment:
Please incorporate the "path" argument into the error message. This will help the user/developer better diagnose what is going wrong.